marlowe-playground-client/src/Simulation/BottomPanel.purs

```diff
@@ -0,0 +1,495 @@
+            <> show owner
+            <> " to "
+            <> ( case payee of
+                  (Account (AccountId accNum2 owner2)) -> "account " <> show accNum2 <> " of " <> show owner2
+                  (Party dest) -> "party " <> show dest
+              )
+            <> "."
+        ]
+    ]
+
+  displayWarning' (TransactionPartialPay (AccountId accNum owner) payee tok amount expected) =
+    [ div [ classes [ rTableCell, first ] ] []
+    , div [ class_ (ClassName "RTable-2-cells") ] [ text "TransactionPartialPay" ]
+    , div [ class_ (ClassName "RTable-4-cells") ]
+        [ text $ "The contract is supposed to make a payment of "
+            <> show expected
+            <> " units of "
+            <> show tok
+            <> " from account "
+            <> show accNum
+            <> " of "
+            <> show owner
+            <> " to "
+            <> ( case payee of
+                  (Account (AccountId accNum2 owner2)) -> ("account " <> show accNum2 <> " of " <> show owner2)
+                  (Party dest) -> ("party " <> show dest)
+              )
+            <> " but there is only "
+            <> show amount
+            <> "."
+        ]
+    ]
+
+  displayWarning' (TransactionShadowing valId oldVal newVal) =
+    [ div [ classes [ rTableCell, first ] ] []
+    , div [ class_ (ClassName "RTable-2-cells") ] [ text "TransactionShadowing" ]
+    , div [ class_ (ClassName "RTable-4-cells") ]
+        [ text $ "The contract defined the value with id "
+            <> show valId
+            <> " before, it was assigned the value "
+            <> show oldVal
+            <> " and now it is being assigned the value "
+            <> show newVal
+            <> "."
+        ]
+    ]
+
+panelContents state StaticAnalysisView =
+  section
+    [ classes [ ClassName "panel-sub-header", aHorizontal, Classes.panelContents ]
+    ]
+    [ analysisResultPane state
+    , button [ onClick $ const $ Just $ AnalyseContract, enabled (state ^. _analysisState <<< to isLoading <<< to not) ] [ text "Analyse" ]
+    ]
+
+panelContents state MarloweWarningsView =
+  section
+    [ classes [ ClassName "panel-sub-header", aHorizontal, Classes.panelContents, flexLeft ]
+    ]
+    (map renderWarning (state ^. (_marloweState <<< _Head <<< _editorWarnings)))
+  where
+  renderWarning warning = pre [ class_ (ClassName "warning-content") ] [ text warning.text ]
+
+panelContents state MarloweErrorsView =
+  section
+    [ classes [ ClassName "panel-sub-header", aHorizontal, Classes.panelContents, flexLeft ]
+    ]
+    (map renderError (state ^. (_marloweState <<< _Head <<< _editorErrors)))
+  where
+  renderError error = pre [ class_ (ClassName "error-content") ] [ text error.text ]
+
+analysisResultPane :: forall p. FrontendState -> HTML p HAction
+analysisResultPane state =
+  let
+    result = state ^. _analysisState
+  in
+    case result of
+      NotAsked ->
+        div [ classes [ ClassName "padded-explanation" ] ]
```

Comment:
Time to lift `div [ classes [ ClassName "padded-explanation" ] ]` into a `let`? Or at least the classname`?

Reply:
done. I was unsure about 'componentizing' (moving into functions) HTML stuff as I thought it might make it harder for non-purescript folks to read but at this point I'm not concerned with that any more.